nuxeo-core/nuxeo-core-lock/src/main/java/org/nuxeo/ecm/core/lock/LockHelper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ *
+ */
+
+package org.nuxeo.ecm.core.lock;
+
+import org.nuxeo.ecm.core.api.ConcurrentUpdateException;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.kv.KeyValueService;
+import org.nuxeo.ecm.core.kv.KeyValueStore;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.transaction.TransactionHelper;
+
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Utilities to work with locks
+ *
+ * @since 9.3
+ */
+public class LockHelper {
+
+    public static final String DOCUMENT_LOCK = "document-lock";
+
+    public static final String LOCK = "lock";
+
+    public static final int NB_TRY = 3;
+
+    public static final int SLEEP_DURATION = 1000;
+
+    private LockHelper() {
+    }
+
+    /**
+     * Performs a {@link Function} atomically, in a cluster-wide critical section.
+     *
+     * @param key the key
```

Comment:
Explain a bit more how the key is used, like: "the key used to determine atomicity".